test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ModelStateDictionaryExtensionsTest.cs

```diff
@@ -0,0 +1,336 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class ModelStateDictionaryExtensionsTest
+    {
+        [Fact]
+        public void AddModelError_ForSingleExpression_AddsExpectedMessage()
+        {
+            // Arrange
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            modelState.AddModelError<TestModel>(model => model.Text, "Message");
+
+            // Assert
+            Assert.Equal("Text", modelState.Single().Key);
```

Comment:
FYI only: `Assert.Single()` results in significantly better test failures than the `Exception` Linq's `Single()` throws. If you're kind enough to tackle other issues in the future, do this as `Assert.Equal("Text", Assert.Single(modelState).Key);`. Or use a temporary variable for the `Assert.Single()` result so that each assertion is on a single line and the exact failure is easier to find.